tests/modules/transformer/self_attention_test.py

```diff
@@ -1,20 +1,84 @@
 import copy
 import torch
+import pytest
 
 from allennlp.common import Params
+from allennlp.common import cached_transformers
+from allennlp.common.testing import assert_equal_parameters
+
 from allennlp.modules.transformer import SelfAttention
 from allennlp.common.testing import AllenNlpTestCase
 
+from transformers.configuration_bert import BertConfig
+from transformers.modeling_bert import BertSelfAttention
+from transformers.configuration_roberta import RobertaConfig
+from transformers.modeling_roberta import RobertaSelfAttention
+from transformers.configuration_electra import ElectraConfig
+from transformers.modeling_electra import ElectraSelfAttention
+from transformers.configuration_distilbert import DistilBertConfig
+from transformers.modeling_distilbert import MultiHeadSelfAttention
+
+# from transformers.configuration_mobilebert import MobileBertConfig
+# from transformers.modeling_mobilebert import MobileBertSelfAttention
+# from transformers.configuration_t5 import T5Config
+# from transformers.modeling_t5 import T5LayerSelfAttention
+
+PARAMS_DICT = {
+    "hidden_size": 6,
+    "num_attention_heads": 2,
+    "dropout": 0.0,
+}
+
+
+def get_modules(params_dict):
+    modules = {}
+    params = copy.deepcopy(params_dict)
+    params["attention_probs_dropout_prob"] = params.pop("dropout")
+
+    # bert, roberta, electra, layoutlm self attentions have the same code.
```

Comment:
`layoutlm`? That's not mentioned again later.

Reply:
Yes, I didn't add it because it's the exact same code as bert. Technically, tests for roberta and electra aren't needed either (at least for the self-attention module). I just added them to demonstrate that the toolkit modules are expected to work for different models.